File: dongyimai_seckill_interface/src/main/java/com/seckill/service/SeckillGoodsService.java
package com.seckill.service;
import com.dongyimai.bean.TbSeckillGoods;
import com.dongyimai.result.PageResult;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface SeckillGoodsService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbSeckillGoods> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbSeckillGoods seckill_goods);
	
	
	/**
	 * 修改
	 */
	public void update(TbSeckillGoods seckill_goods);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbSeckillGoods findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbSeckillGoods seckill_goods, int pageNum,int pageSize);

	/**
	 * 返回当前正在参与秒杀的商品
	 * @return
	 */
	public List<TbSeckillGoods> findList();

	/**
	 * 根据ID获取实体(从缓存中读取)
	 */
	public TbSeckillGoods findOneFromRedis(Long id);

}

File: dongyimai_seckill_web/src/main/webapp/js/service/seckillGoodsService.js
//服务层
app.service('seckillGoodsService',function($http){
    //读取列表数据绑定到表单中
    this.findList=function(){
        return $http.get('seckillGoods/findList.do');
    }

    //根据指定id查询秒杀商品
    this.findOne=function(id){
        return $http.get('seckillGoods/findOneFromRedis.do?id='+id);
    }

    //提交订单
    this.submitOrder = function(seckillId){
        return $http.get('seckillOrder/submitOrder.do?seckillId='+seckillId);
    }

    //查询当前登陆的用户名
    this.getLoginUserName = function(){
        return $http.get('seckillGoods/getLoginUserName.do');
    }
});
